tests/jax/sharding_configs.py

```diff
@@ -0,0 +1,101 @@
+import jax
+from dataclasses import dataclass
+from itertools import product
+from transformer_engine.jax.sharding import ShardingType
+from transformer_engine.jax.softmax import SoftmaxType
+from transformer_engine.jax.fused_attn import AttnBiasType, AttnMaskType\
+
+
+class ShardingConfigs(object):
+    
+    def __init__(self, num_gpus=jax.device_count('gpu')):
+        super().__init__()
+        if num_gpus < 2:
+            raise ValueError(f"ShardingConfig: Need at least 2 GPUs, but got {num_gpus}.")
+        
+        self.device_count = min(num_gpus, 8)
+        mesh_configs = [
+            ((self.device_count, 1),    ("dp", None), ShardingType.DP),
+            ((self.device_count, 1),    ("tp", None), ShardingType.TP_COL),
+            ((self.device_count, 1),    ("tp", None), ShardingType.TP_ROW),
+        ]
+        if self.device_count > 2:
```

Comment:
It doesn't assume power of 2 GPUs, but it will always use even numbers to test DP_TP configurations because there's no other way to construct a 2D device mesh. That will leave an unused GPU when device count is 5 or 7. Device count of 6 fits cleanly here with a 3x2 grid even though it's not a power of two.

That said, I see that 3 GPUs are unaccounted for right now. It will try (and fail?) to test DP_TP with a 1x1 grid. I will modify the check to be >3 instead of >2 to avoid that.